### PR TITLE
feat: round progress

### PR DESCRIPTION
**Pull request by evgeniy-ac**

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/OnboardingModal.razor [hunk 1]
```diff
@@ -38,12 +43,34 @@
...
-                Decoration="LineProgress.LineProgressDecoration.Blue">
-            </LineProgress>
+            <div class="flex w-full justify-center mt-6">
+                <LineProgress
+                    TotalSteps="stepper.TotalSteps"
+                    CurrentStep="stepper.CurrentStepIndex"
+                    Decoration="LineProgress.LineProgressDecoration.Blue">
+                </LineProgress>
+            </div>
+            @* RoundProgress example *@
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/Onboarding/OnboardingModal.razor:53]: is still needed?

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+</div>
+
+@code {
+    [Parameter] public int TotalSteps { get; set; }
+    [Parameter] public int CurrentStep { get; set; }
+    [Parameter] public double Diameter { get; set; } = 58;
+    [Parameter] public RenderFragment? ChildContent { get; set; }
+
+    private double Center => Diameter / 2;
+    private double Radius => Diameter / 2 - 2;
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor:36]: why minus two?

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    private double Circumference => 2 * Math.PI * Radius;
+    private double ProgressPercent => TotalSteps == 0 ? 0 : CurrentStep / ((double)TotalSteps / 100);
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor:38]: looks unnatural `part/(whole/100)`. Maybe `(part/whole)*100`?

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    private double Offset => Circumference - ProgressPercent / 100 * Circumference;
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor:39]: looks like it's better to store progress and do all the math with progress, while progressPercent is more like representual property

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/OnboardingModal.razor (hunk 1 above)
Comment (evgeniy-ac) [reply src/dotnet/Chat.UI.Blazor/Components/Onboarding/OnboardingModal.razor:53]: yes, there are no steps which use it

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor (hunk 3 above)
Comment (evgeniy-ac) [reply src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor]: Fixed

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor (hunk 2 above)
Comment (evgeniy-ac) [reply src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor]: Fixed. It's width of circle border

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/OnboardingModal.razor [hunk 5]
```diff
@@ -61,5 +88,9 @@
         await ModalInstance.CloseAsync();
     }
 
+    private async Task OnCloseClick() {
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/Onboarding/OnboardingModal.razor:91]: 1. expression
2. await is redundant

**Review by FrolYo — APPROVED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/Onboarding/OnboardingModal.razor (hunk 5 above)
Comment (evgeniy-ac) [reply src/dotnet/Chat.UI.Blazor/Components/Onboarding/OnboardingModal.razor]: done

**Pull request merged**